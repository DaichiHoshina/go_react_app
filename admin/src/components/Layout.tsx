import React, { useEffect, useRef, useState } from "react";
import CssBaseline from "@material-ui/core/CssBaseline";
import Drawer from "@material-ui/core/Drawer";
import Box from "@material-ui/core/Box";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import MenuIcon from "@material-ui/icons/Menu";
import IconButton from "@material-ui/core/IconButton";
import { useRouter } from "next/router";
import SideMenuArea from "./templates/SideMenuArea";

const drawerWidth = 240;

const Copyright = () => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright © Daichi Hoshina"}
    </Typography>
  );
};

export interface LayoutProps {
  children: React.ReactNode;
  title: string;
}

const Layout: React.FC<LayoutProps> = ({ children, title }) => {
  const [isOpen, setIsOpen] = useState(true);
  const sideMenuRef = useRef<HTMLDivElement>(null);
  const router = useRouter();
  const [sideMenuWidth, setSideMenuWidth] = useState(0);

  const handleDrawerOpen = () => {
    setIsOpen(!isOpen);
  };

  useEffect(() => {
    if (sideMenuRef.current !== null) {
      setSideMenuWidth(sideMenuRef.current.clientWidth);
    }
  });

  return (
    <div className="min-h-screen bg-blue-50 pt-10">
      <CssBaseline />
      <AppBar style={{ zIndex: 1201 }} position="fixed">
        <Toolbar variant="dense">
          <IconButton
            edge="start"
            color="inherit"
            aria-label="menu"
            onClick={handleDrawerOpen}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" color="inherit" className="flex-grow">
            管理画面
          </Typography>
        </Toolbar>
      </AppBar>
      <div
        style={{
          paddingLeft:
            isOpen && !router.pathname.includes("/login") ? sideMenuWidth : 0,
        }}
      >
        <div className="p-7">
          <Typography component="h2" variant="h6" color="inherit">
            {title}
          </Typography>
          {children}
        </div>
      </div>
      <Box className="pt-8">
        <Copyright />
      </Box>
      <Drawer open={isOpen} variant="persistent" anchor="left">
        <div ref={sideMenuRef} />
        <SideMenuArea />
      </Drawer>
    </div>
  );
};

export default Layout;
