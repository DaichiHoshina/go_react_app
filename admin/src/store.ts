import { combineReducers } from 'redux';
import logger from 'redux-logger';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { userSlice, initialState as userInitialState } from './modules/User';

const rootReducer = combineReducers({
  userState: userSlice.reducer,
});

const preloadedState = () => {
  return { userState: userInitialState };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== 'production',
    preloadedState: preloadedState(),
  });
};

export default store;
