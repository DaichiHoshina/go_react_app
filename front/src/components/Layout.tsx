import React, { useEffect, useState } from "react";
import CssBaseline from "@material-ui/core/CssBaseline";
import Box from "@material-ui/core/Box";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Footer from "./organisms/common/Footer";
import { Link } from "@material-ui/core";
import { TUser } from "../modules/User";
import HeaderUserIcon from "../components/atoms/share/HeaderUserIcon";
import { loginConfirm } from "../services/User";
import { useDispatch } from "react-redux";
import LoginIcon from "./atoms/share/LoginIcon";
export interface LayoutProps {
  children?: React.ReactNode;
  title: string;
}

export const Layout: React.FC<LayoutProps> = ({ children, title }) => {
  const dispatch = useDispatch();
  const [isLogin, setIsLogin] = useState(false);

  // ログイン中か確認する
  const userLoginConfirm = async () => {
    const result: any = await dispatch(loginConfirm());
    const user: TUser = result.payload.data;
    setIsLogin(!!user);
  };

  useEffect(() => {
    userLoginConfirm();
  }, []);

  return (
    <div className="min-h-screen pt-10">
      <CssBaseline />
      <AppBar
        position="fixed"
        style={{ zIndex: 1201, color: "#FFFFFF", backgroundColor: "#000000" }}
      >
        <div className="flex">
          <div className="mt-2">
            <Toolbar variant="dense">
              <Link href="/presentations" color="inherit" underline="none">
                {/* タイトルロゴ */}
                <Typography variant="h6" color="inherit" className="flex-grow">
                  Repgram
                </Typography>
              </Link>
            </Toolbar>
          </div>
          {/* 右上のアイコン */}
          <div className="ml-auto mt-2 mr-2">
            {isLogin ? <HeaderUserIcon /> : <LoginIcon />}
          </div>
        </div>
      </AppBar>
      <div className="p-7 mt-3">
        <Typography component="h2" variant="h6" color="inherit">
          {title}
        </Typography>
        <div>{children}</div>
      </div>
      <Box className="pt-8">
        <Footer />
      </Box>
    </div>
  );
};

export default Layout;
