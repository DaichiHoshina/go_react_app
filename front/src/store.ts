import { combineReducers } from "redux";
import logger from "redux-logger";
import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import { userSlice, initialState as userInitialState } from "./modules/User";
import {
  presentationSlice,
  initialState as presentationInitialState,
} from "./modules/Presentation";

const rootReducer = combineReducers({
  userState: userSlice.reducer,
  presentationState: presentationSlice.reducer,
});

const preloadedState = () => {
  return {
    userState: userInitialState,
    presentationState: presentationInitialState,
  };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== "production",
    preloadedState: preloadedState(),
  });
};

export default store;
