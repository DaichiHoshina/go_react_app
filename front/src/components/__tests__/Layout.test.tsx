import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../store";
import Layout from "../Layout";

it("<Layout>のスナップショット", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <Layout title="" />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
