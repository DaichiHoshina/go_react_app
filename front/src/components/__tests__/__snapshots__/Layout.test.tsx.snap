// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout>のスナップショット 1`] = `
<div
  className="min-h-screen pt-10"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "#FFFFFF",
        "zIndex": 1201,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="mt-2"
      >
        <div
          className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="/presentations"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <h6
              className="MuiTypography-root flex-grow MuiTypography-h6 MuiTypography-colorInherit"
            >
              Repgram
            </h6>
          </a>
        </div>
      </div>
      <div
        className="ml-auto mt-2 mr-2"
      >
        <div
          className="flex"
        >
          <div
            className="mr-2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign In
              </span>
            </button>
          </div>
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign Up
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="p-7 mt-3"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    >
      
    </h2>
    <div />
  </div>
  <div
    className="MuiBox-root MuiBox-root-1 pt-8"
  >
    <div
      className="mt-5 pt-5 text-center text-xs border-t border-gray-300"
    >
      Copyright 2021 Daichi Hoshina
    </div>
  </div>
</div>
`;
